boot/bootutil/src/loader.c

```diff
@@ -1286,6 +1278,15 @@ boot_go(struct boot_rsp *rsp)
         break;
     }
 
+#ifdef MCUBOOT_VALIDATE_SLOT0
+    rc = boot_validate_slot(0);
```

Comment:
I'm not sure about this. Doesn't this need to re-read the image headers? boot_validate_slot() just uses the cached headers in boot_data, which were read before any swapping was done.